presto-verifier/src/main/java/com/facebook/presto/verifier/framework/AbstractVerification.java

```diff
@@ -475,6 +485,13 @@ private Optional<SkippedReason> getSkippedReason(Optional<Throwable> throwable,
         if (skipControl && !QUERY_BANK_MODE.equals(runningMode)) {
             return Optional.empty();
         }
+        if (rewriteNonDeterministicFunctions && NOT_RUN.equals(controlState) && throwable
```

Comment:
Why is the controlStateCheck done here ? Is there a way to know that the rewrite was done ? I guess both control and test querier are rewritten right ?

Reply:
This is in the path after the control and perhaps the test query fail. Here I use `rewriteNonDeterministicFunctions` = verifierConfig.getNonDeterministicFunctionSubstitutes().isPresent() to tell if rewrite was done.

As you might know we only need to look at control to decide if the test case should be skipped. If the control query fails, it means the test case is somehow invalid and we want to skip it. As we rewrite the function calls of the control query and the test query the same way, if the control query is invalid, then we can say the test case is invalid and we should skip.